src/mmw/js/src/modeling/models.js

```diff
@@ -130,12 +164,32 @@ var ScenarioModel = Backbone.Model.extend({
         name: '',
         is_current_conditions: false,
         modifications: null,      // ModificationsCollection
-        active: false
+        active: false,
+        uid: null
     },
 
     initialize: function(attrs, options) {
         Backbone.Model.prototype.initialize.apply(this, arguments);
+        this.set('uid', App.user.get('uid'));
         this.set('modifications', ModificationsCollection.create(attrs.modifications));
+        this.on({
+            'change:project': this.attemptSave,
+            'change:name': this.attemptSave
+        }, this);
+
+        this.get('modifications').on({
+            'add': this.attemptSave,
+            'remove': this.attemptSave
+        }, this);
+    },
+
+    attemptSave: function() {
+        if (App.user.get('uid') !== this.get('uid')) {
```

Comment:
Right good. catch. I'm going to make the API endpoint return a uid of zero for guest users and then update the checks for user matching to ensure a non zero and matching uid.